services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Persistence.kt

```diff
@@ -179,5 +180,17 @@ interface HankeRepository : JpaRepository<HankeEntity, Int> {
 
     override fun findAll(): List<HankeEntity>
 
-    // TODO: add any special 'find' etc. functions here, like searching by date range.
+    // search with date range
+    fun findAllByAlkuPvmIsBeforeAndLoppuPvmIsAfter(endAlkuPvm: LocalDate, startLoppuPvm: LocalDate): List<HankeEntity>
```

Comment:
Does this work out-of-the-box like this? JPA creates the query automatically based on function name? Cool! 
What is the commented-out function below ("getAllDataHankeBetweenTimePeriod")?

Reply:
Jep. I think shortly mentioned it during the JPA super-short presentation, but yeah, too shortly.

It does have its limits, in which case the Query stuff can be used.  The commented part was left as an example from an earlier version (discussed between me and Tiina).

Reply:
Yeah! Pretty neat!
Commented out is an example for the query way of doing the same. My first thought was that the clause will be more difficult and the automatic way would not be enough. Markku helped with the algorithm and it was actually so simple that the out-of-the-box way seems to work. 
Did not want to loose the example though so it is there if someone else needs to find out how the query needs to be added. Maybe saves couple of minutes of searching the solution.